internal/gps/result.go

```diff
@@ -61,19 +65,45 @@ func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool) error {
 		return err
 	}
 
-	// TODO(sdboyer) parallelize
-	for _, p := range l.Projects() {
-		to := filepath.FromSlash(filepath.Join(basedir, string(p.Ident().ProjectRoot)))
+	projects := l.Projects()
+	worksz := len(projects)
+
+	workChan := make(chan LockedProject, worksz)
+	doneChan := make(chan error, worksz)
+
+	for i := 0; i < NumDepTreeWorkers; i++ {
+		go func() {
+			for p := range workChan {
+				to := filepath.FromSlash(filepath.Join(basedir, string(p.Ident().ProjectRoot)))
 
-		err = sm.ExportProject(p.Ident(), p.Version(), to)
+				err := sm.ExportProject(p.Ident(), p.Version(), to)
+				if err != nil {
+					doneChan <- fmt.Errorf("error while exporting %s: %s", p.Ident().ProjectRoot, err)
+					continue
+				}
+
+				if sv {
+					filepath.Walk(to, stripVendor)
+				}
+
+				doneChan <- nil
+			}
+			// TODO(sdboyer) dump version metadata file
+		}()
+	}
+
+	for _, p := range projects {
+		workChan <- p
+	}
+
+	close(workChan)
+
+	for i := 0; i < worksz; i++ {
+		err := <-doneChan
```

Comment:
as-written, this could interleave poorly:

1. one goroutine sends experiences and sends an error while other(s) still writing
2. control receives the error, and begins `os.RemoveAll()`
3. other goroutine(s) are trying to write to dirs that are disappearing from underneath them

additionally, it's also crucial that we guarantee the worker goroutines have terminated by the time the `WriteDepTree()` func returns.

switching to an unbuffered channel approach will, i think, make it easier to do the necessary flow control here.